ddtrace/tracer/textmap.go

```diff
@@ -794,13 +794,71 @@ func (*propagatorW3c) injectTextMap(spanCtx ddtrace.SpanContext, writer TextMapW
 	return nil
 }
 
+// stringMutator maps characters in a string to new characters. It is a state machine intended
+// to replace regex patterns for simple character replacement, including collapsing runs of a
+// specific range.
+//
+// It's designed after the `hash#Hash` interface, and to work with `strings.Map`.
+type stringMutator struct {
+	// n is the current state of the mutator. It is used to track runs of characters that should
+	// be collapsed.
+	n int
```

Comment:
I was a little confused by `n` being an integer. It looks like it only ever takes two values, 0 or -1. Maybe better as a bool?

Reply:
My reasoning at some point was that I wanted to decrease the `n` state as much as consecutive characters I found. But later I just decided to flip it, so a `bool` is better suited for the current implementation.